tests/lib/tools/query-mondodb

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import os
+import sys
+
+from pymongo import MongoClient
+
+
+def _make_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--host', help="Hostname or ip for the mongo database",
+                        **environ_or_required('MONGO_HOST'))
+    parser.add_argument('--port', help="Port for the mongo database",
+                        **environ_or_required('MONGO_PORT'))
+    
+    parser.add_argument('--db-name', help="Name for the database used to insert", default="snapd")
+    parser.add_argument('--db-collection', help="Name for the collection used to insert", default="spread_errors")
+    parser.add_argument('--after-date', help="Find results after a date (yyyy-mm-dd)", default="2000-01-01")
+    parser.add_argument('--after-time', help="Find results after a time (hh:mm:ss)", default="00:00:00")
+    parser.add_argument('--before-date', help="Find results before a date (yyyy-mm-dd)", default="2999-12-12")
+    parser.add_argument('--before-time', help="Find results before a time (hh:mm:ss)", default="24:00:00")
+    parser.add_argument('--system', help="Find results for a specific spread system", default=".*")
+    parser.add_argument('--backend', help="Find results for a specific spread backend", default=".*")
+    parser.add_argument('--task', help="Find results for a specific spread task (include variant)", default=".*")
+    parser.add_argument('--include', help="Value to include in the output", action="append", default=[])
+    parser.add_argument('--exclude', help="Value to exclude from the output", action="append", default=[])
```

Comment:
I'm not clear what include/exclude are about, I suppose explanation is too long for the help string here, maybe a comment in the code describing them?

Reply:
Yeah a help text for this tool would help to review this, I think from what Sergio has described about the purpose of this tool, include and exclude are essentially row names for i.e. individual tests? so you can say give me all test results except for this specific test I know is bad, or give me all test results just for this specific test, etc

Reply:
this perhaps is a bit confusing, include and exclude are the projections. 
"--include  A --include B" is like when do "SELECT A,B" in a SQL
Please suggest a new command and I'll change it. I added this because sometimes you don't want to see the error output, you just want to see which tests failed.